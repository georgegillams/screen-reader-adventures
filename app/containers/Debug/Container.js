import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import Helmet from 'react-helmet';
import { cssModules } from 'bpk-react-utils';
import { getTimeDifference } from 'helpers/time';
import { Checkbox } from 'gg-components/Checkbox';

import STYLES from '../pages.scss';

import { Paragraph, PageTitle } from 'gg-components/Typography';
const getClassName = cssModules(STYLES);

const StatusControl = props => {
  const { name, storageKey, ...rest } = props;
  const [isEnabled, setIsEnabled] = useState(false);
  useEffect(() => {
    const isEnabled = window.localStorage.getItem(storageKey) === 'true';
    setIsEnabled(isEnabled);
  }, []);

  const onValueChanged = newValue => {
    setIsEnabled(newValue);
    window.localStorage.setItem(storageKey, newValue ? 'true' : 'false');
  };

  return (
    <Checkbox
      label={name}
      checked={isEnabled}
      onChange={event => {
        onValueChanged(event.target.checked);
      }}
      {...rest}
    />
  );
};

const Debug = props => {
  return (
    <div {...props}>
      <Helmet title="Debug" />
      <PageTitle anchor={false} name="Debug">
        <StatusControl
          name="Show session debug views"
          storageKey="showSessionDebugViews"
        />
        <br />
        <br />
        <StatusControl
          name="Show page container debug colours"
          storageKey="showPageContainerDebugColor"
        />
        <br />
        <Paragraph>
          Note that changes will not take effect until you reload the page.
        </Paragraph>
      </PageTitle>
    </div>
  );
};

export default Debug;
