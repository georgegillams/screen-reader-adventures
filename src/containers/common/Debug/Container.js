import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { Checkbox } from 'gg-components/Checkbox';
import { Paragraph } from 'gg-components/Paragraph';
import PageTitle from 'components/common/PageTitle';
import STYLES from './debug.scss';
import { cssModules } from 'gg-components/helpers/cssModules';
import { DEBUG_SHOW_PAGE_CONTAINER_KEY, DEBUG_SHOW_DEBUG_INFORMATION_KEY } from 'helpers/storageConstants';

const getClassName = cssModules(STYLES);

const StatusControl = props => {
  const { name, storageKey, ...rest } = props;
  const [isEnabled, setIsEnabled] = useState(false);

  useEffect(() => {
    const debugEnabled = window.localStorage.getItem(storageKey) === 'true';
    setIsEnabled(debugEnabled);
  }, []);

  const onValueChanged = newValue => {
    setIsEnabled(newValue);
    window.localStorage.setItem(storageKey, newValue ? 'true' : 'false');
  };

  return (
    <Checkbox
      name={name}
      label={name}
      checked={isEnabled}
      onChange={event => {
        onValueChanged(event.target.checked);
      }}
      {...rest}
    />
  );
};

StatusControl.propTypes = {
  name: PropTypes.string.isRequired,
  storageKey: PropTypes.string.isRequired,
};

const Debug = props => (
  <div {...props}>
    <PageTitle anchor={false} name="Debug">
      <StatusControl
        className={getClassName('debug__checkbox')}
        name="Show session debug views"
        storageKey={DEBUG_SHOW_DEBUG_INFORMATION_KEY}
      />
      <StatusControl
        className={getClassName('debug__checkbox')}
        name="Show page container debug colours"
        storageKey={DEBUG_SHOW_PAGE_CONTAINER_KEY}
      />
      <Paragraph>Note that changes will not take effect until you reload the page.</Paragraph>
    </PageTitle>
  </div>
);

export default Debug;
