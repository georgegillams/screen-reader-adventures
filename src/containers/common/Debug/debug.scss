@import '~utils/tokens';

.debug {
  &__checkbox {
    margin-top: $spacing-base;
    margin-bottom: $spacing-base;
  }
}
