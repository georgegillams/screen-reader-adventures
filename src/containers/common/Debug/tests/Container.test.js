import React from 'react';
import { render } from '@testing-library/react';

import Debug from '../Container';

describe('<Debug />', () => {
  it('should render correctly', () => {
    const { container } = render(<Debug />);

    expect(container).toMatchSnapshot();
  });
});
