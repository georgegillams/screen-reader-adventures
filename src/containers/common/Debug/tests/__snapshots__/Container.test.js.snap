// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Debug /> should render correctly 1`] = `
<div>
  <div>
    <h1
      class="page-title__heading"
    >
      Debug
    </h1>
    <label
      class="checkbox debug__checkbox"
    >
      <div
        class="checkbox__input-wrapper"
      >
        <input
          aria-invalid="false"
          aria-label="Show session debug views"
          class="checkbox__input"
          name="Show session debug views"
          type="checkbox"
        />
        <svg
          class="icon checkbox__check"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.4 20c-.5 0-.9-.2-1.3-.5l-5.8-5.1c-.4-.4-.5-1-.1-1.4l1.3-1.5c.4-.4 1-.5 1.4-.1l4 3.5c.2.1.4.1.6 0l9.2-10.5c.4-.4 1-.5 1.4-.1l1.5 1.3c.4.4.5 1 .1 1.4L10.9 19.3c-.4.5-.9.7-1.5.7z"
          />
        </svg>
      </div>
      <span
        aria-hidden="true"
        class="checkbox__label"
      >
        Show session debug views
         
      </span>
    </label>
    <label
      class="checkbox debug__checkbox"
    >
      <div
        class="checkbox__input-wrapper"
      >
        <input
          aria-invalid="false"
          aria-label="Show page container debug colours"
          class="checkbox__input"
          name="Show page container debug colours"
          type="checkbox"
        />
        <svg
          class="icon checkbox__check"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.4 20c-.5 0-.9-.2-1.3-.5l-5.8-5.1c-.4-.4-.5-1-.1-1.4l1.3-1.5c.4-.4 1-.5 1.4-.1l4 3.5c.2.1.4.1.6 0l9.2-10.5c.4-.4 1-.5 1.4-.1l1.5 1.3c.4.4.5 1 .1 1.4L10.9 19.3c-.4.5-.9.7-1.5.7z"
          />
        </svg>
      </div>
      <span
        aria-hidden="true"
        class="checkbox__label"
      >
        Show page container debug colours
         
      </span>
    </label>
    <span
      class="paragraph"
    >
      Note that changes will not take effect until you reload the page.
    </span>
  </div>
</div>
`;
